armory/scenarios/base.py

```diff
@@ -41,11 +43,18 @@ def evaluate(self, config: dict):
         if results is None:
             logger.warning(f"{self._evaluate} returned None, not a dict")
         self.save(config, results)
+        self.teardown()
 
     def setup(self, config: dict):
         """
         Performs scenario setup on the evaluating system.
         """
+
+        id = uuid.uuid4()
+        self.scenario_output_dir = os.path.join(paths.docker().output_dir, str(id))
+        os.mkdir(self.scenario_output_dir)
+        logger.warning(f"Outputs will be written to {self.scenario_output_dir}")
+
         if config["sysconfig"].get("external_github_repo", None):
             external_repo_dir = paths.docker().external_repo_dir
             external_repo.download_and_extract_repo(
```

Comment:
Suggestion here:
Have `paths` handle the creation of output and tmp directories the first time `paths.docker()` is called. Since this is then cached, all the other code shouldn't need changing. The tmp and output subdirs could just be returned directly from paths.

Reply:
I'm in favor of directory creation being explicit to the someone reading the scenario code instead of done behind the scenes when accessing paths. I'm planning on submitting a paths refactor today. Perhaps we could discuss in that PR?

Reply:
That works for me.